aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,241 @@
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient<GitHubService>();
+    services.AddMvc();
+}
+```
+
+The typed client is registered as transient with the DI framework.
+
+The typed client can then be injected and consumed directly. 
+
+```csharp
+public class MyController : Controller
+{
+    private GitHubService _gitHubService;
+
+    public MyController(GitHubService gitHubService)
+    {
+        _gitHubService = gitHubService;
+    }
+
+    public IActionResult Index()
+    {
+        var result = await _gitHubService.Client.GetStringAsync("/orgs/octokit/repos");
+        return Ok(result);
+    }
+}
+```
+
+In this example we only moved configuration into the type, but we could also have methods with behaviour and not actually expose the HttpClient if we want all access to go through this type. 
+
+If you prefer, the configuration for a typed client can be specified during registration, rather than in the constructor for the typed client.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient<GitHubService>(c =>
+    {
+        c.BaseAddress = new Uri("https://api.github.com/");
+
+        c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // Github API versioning
+        c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // Github requires a user-agent
+    });
+    services.AddMvc();
+}
+```
+
+If we want to entirely encapsulate the HttpClient in our typed client, rather than exposing it as a property we can define our own methods which control the client
+
+```csharp
+public class ValuesService
+{
+    private readonly HttpClient _httpClient;
+    private readonly IMemoryCache _cache;
+    private readonly ILogger<ValuesService> _logger;
+
+    public ValuesService() { }
+
+    public ValuesService(HttpClient client, IMemoryCache cache, ILogger<ValuesService> logger)
+    {
+        _httpClient = client;
+        _cache = cache;
+        _logger = logger;
+    }
+
+    public async Task<IEnumerable<string>> GetValues()
+    {
+        var result = await _httpClient.GetAsync("api/values");
+        var resultObj = Enumerable.Empty<string>();
+
+        if (result.IsSuccessStatusCode)
+        {
+            resultObj = JsonConvert.DeserializeObject<IEnumerable<string>>(await result.Content.ReadAsStringAsync());
+            _cache.Set("GetValue", resultObj);
+        }
+        else
+        {
+            if (_cache.TryGetValue("GetValue", out resultObj))
+            {
+                _logger.LogWarning("Returning cached values as the values service is unavailable.");
+                return resultObj;
+            }
+            result.EnsureSuccessStatusCode();
+        }        
+        return resultObj;
+    }
+}
+```
+
+In this example the HttpClient is stored as a private field and all access to make external calls goes through the GetValues method.
+
+## Generated clients
+
+TODO - Refit?
```

Comment:
Refit integration demo is here: https://github.com/rynowak/HttpClientFactoryDemos/tree/master/RefitSample

Reply:
Added, thanks!